core/src/test/java/net/sourceforge/jnlp/util/whitelist/WhitelistEntryTest.java

```diff
@@ -0,0 +1,115 @@
+        final WhitelistEntry invalidEntry = validateWhitelistUrl("");
+
+        assertInvalid(invalidEntry);
+
+        assertNoMatch(invalidEntry, null);
+        assertNoMatch(invalidEntry, "http://localhost");
+        assertNoMatch(invalidEntry, "https://192.168.1.1");
+    }
+
+    @Test
+    public void exactEntryShouldOnlyMatchExactUrl() throws Exception {
+        final WhitelistEntry validEntry = validateWhitelistUrl("http://test.com:88");
+
+        assertMatch(validEntry, "http://test.com:88");
+
+        assertNoMatch(validEntry, "https://test.com:88"); // wrong protocol
+        assertNoMatch(validEntry, "http://test2.com:88"); // wrong host
+        assertNoMatch(validEntry, "http://test.com:888"); // wrong port
+    }
+
+    @Test
+    public void defaultPortShouldMatchExplicitAndAbsentValue() throws Exception {
+        final WhitelistEntry explicitHttpEntry = validateWhitelistUrl("http://test.com:80");
+        final WhitelistEntry implicitHttpEntry = validateWhitelistUrl("http://test.com");
+        final WhitelistEntry explicitHttpsEntry = validateWhitelistUrl("https://test.com:443");
+        final WhitelistEntry implicitHttpsEntry = validateWhitelistUrl("https://test.com");
+
+        assertMatch(explicitHttpEntry, "http://test.com:80");
+        assertMatch(explicitHttpEntry, "http://test.com");
+        assertMatch(implicitHttpEntry, "http://test.com:80");
+        assertMatch(implicitHttpEntry, "http://test.com");
+
+        assertNoMatch(explicitHttpEntry, "http://test.com:88"); // wrong port
+        assertNoMatch(implicitHttpEntry, "http://test.com:88"); // wrong port
+
+        assertMatch(explicitHttpsEntry, "https://test.com:443");
+        assertMatch(explicitHttpsEntry, "https://test.com");
+        assertMatch(implicitHttpsEntry, "https://test.com:443");
+        assertMatch(implicitHttpsEntry, "https://test.com");
+
+        assertNoMatch(explicitHttpsEntry, "https://test.com:444"); // wrong port
+        assertNoMatch(implicitHttpsEntry, "https://test.com:444"); // wrong port
+    }
+
+    @Test
+    public void wildcardPortShouldMatchAnyPort() throws Exception {
+        final WhitelistEntry validEntry = validateWhitelistUrl("http://test.com:*");
+
+        assertMatch(validEntry, "http://test.com");
+        assertMatch(validEntry, "http://test.com:80");
+        assertMatch(validEntry, "http://test.com:443");
+        assertMatch(validEntry, "http://test.com:1");
+        assertMatch(validEntry, "http://test.com:10");
+        assertMatch(validEntry, "http://test.com:100");
+        assertMatch(validEntry, "http://test.com:1000");
+        assertMatch(validEntry, "http://test.com:10000");
+    }
+
+    @Test
+    public void wildcardHostPartShouldMatchAnyPrefix() throws Exception {
+        final WhitelistEntry validEntry = validateWhitelistUrl("http://*.test.com");
+
+        assertMatch(validEntry, "http://sub.test.com");
+        // FIXME: assertMatch(validEntry, "http://sub.sub.test.com");
```

Comment:
@hendrikebbers 
Any comment on this FIXME?

Reply:
we just need to document that only 1 subdomain is supported for now. @janakmulani is this part of the doc?

Reply:
yes